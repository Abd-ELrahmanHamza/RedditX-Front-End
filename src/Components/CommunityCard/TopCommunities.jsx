import { CommunityContainer } from "./TopCommunities.styled";
import { Link } from "react-router-dom";
import CommunityCardItem from "../CommunityCardItemExplore/CommunityCardItemExplore";
import axios from "API/axios";
import useFetch from "Hooks/useFetch";
/**
 * Component that contains the Community Card Elements.
 *
 * @Component
 * @returns {React.Component}
 */
const TopCommunities = ({ CommunityList, communityListSub }) => {
  // let [CommunityList, error, loading, reload] = useFetch({
  //   axiosInstance: axios,
  //   method: "GET",
  //   url: "http://localhost:8000/communityList",
  //   requestConfig: {
  //     headers: {
  //       "Content-Language": "en-US",
  //     },
  //   },
  // });
  if (CommunityList.results && communityListSub) {
    const arr = CommunityList.results;
    console.log(arr);
    const communitiesNumber = CommunityList.results.length;
    return (
      <CommunityContainer>
        <p className="title">Top Communities</p>

        {communitiesNumber !== 0 &&
          CommunityList &&
          CommunityList.results &&
          communityListSub &&
          communityListSub.communities &&
          CommunityList.results.slice(0, 8).map((Community) => (
            <CommunityCardItem
              communityIcon={Community.icon}
              key={Community._id}
              communityID={Community._id}
              communityDescription={Community.description}
              membersCount={Community.membersCnt}
              communityName={Community._id}
              isJoined={communityListSub.communities.find((element) => {
                return element._id === Community._id;
              })}
            />
          ))}
        {communitiesNumber === 0 && <p className="NoResults">No results</p>}
        {communitiesNumber !== 0 && (
          <Link to="/explore/Communities">
            <p className="more">See more</p>
          </Link>
        )}
      </CommunityContainer>
    );
  }
};

export default TopCommunities;
