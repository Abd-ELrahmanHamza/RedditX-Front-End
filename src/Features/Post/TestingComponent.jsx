// Import contexts providers
import { SubmitDestinationProvider } from "Features/Post/Contexts/submitDestination";
import { CreatePostTitleProvider } from "Features/Post/Contexts/createPostTitle";
import { CreatePostFlagsProvider } from "Features/Post/Contexts/createPostFlags";
import { CreatePostAttachmentsProvider } from "Features/Post/Contexts/createPostAttachments";
import { CreatePostFlairsProvider } from "Features/Post/Contexts/createPostFlairs";
import { CreatePostTextProvider } from "Features/Post/Contexts/createPostText";
import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  const auth = useAuth();
  useEffect(() => {
    auth.login({
      expiresIn: 86400,
      token:
        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbFR5cGUiOiJiYXJlIGVtYWlsIiwidXNlcm5hbWUiOiJ0Ml9ob21lIiwiaWF0IjoxNjY5OTI0MDEzLCJleHAiOjE2NzAzNTYwMTN9.FDa59vJi5gWy4dQVkEtgyCfm3Pp2BtHkPz0_AfNVKgY",
      username: "t2_home",
    });
  }, []);

  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SubmitDestinationProvider>
          <CreatePostTitleProvider>
            <CreatePostFlagsProvider>
              <CreatePostAttachmentsProvider>
                <CreatePostFlairsProvider>
                  <CreatePostTextProvider>{children}</CreatePostTextProvider>
                </CreatePostFlairsProvider>
              </CreatePostAttachmentsProvider>
            </CreatePostFlagsProvider>
          </CreatePostTitleProvider>
        </SubmitDestinationProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
