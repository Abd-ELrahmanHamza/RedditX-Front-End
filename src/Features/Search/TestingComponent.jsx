// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";
// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { SafeContextProvider } from "./Contexts/SafeSearchContext/Safe-context";
import { SearchContextProvider } from "./Contexts/SearchWordContext/Search-context";

const TestingComponent = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SearchContextProvider>
          <SafeContextProvider>{children}</SafeContextProvider>
        </SearchContextProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

export default TestingComponent;
