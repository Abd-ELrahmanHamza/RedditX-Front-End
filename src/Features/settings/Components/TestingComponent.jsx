import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        {children}
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
