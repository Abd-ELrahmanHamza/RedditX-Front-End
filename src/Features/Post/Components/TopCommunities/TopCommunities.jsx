import { CommunityContainer } from "./TopCommunities.styled";
import CommunityCardItem from "../CommunityCardItem/CommunityCardItem";
import community from "../../../Subreddit/Components/Community/Community";

const CommunityList = [
  {
    communityID: "1",
    communityDescription: "we are looking for a new community",
    membersCount: "4",
    communityName: "Cufe",
  },
  {
    communityID: "2",
    communityDescription: "we are looking for a new modal",
    membersCount: "1",
    communityName: "ASU",
  },
  {
    communityID: "3",
    communityDescription: "we are looking for a new school",
    membersCount: "1",
    communityName: "MCE",
  },
];
/**
 * Component that contains the community card elements (container for community card elements).
 *
 * @Component
 * @param {object} CommunityList -  contains the Community list elements
 * @param {object} CommunitiesSub2 -  contains the Community list elements that user subscribe
 * @returns {React.Component}
 */
const TopCommunities = () => {
  return (
    <CommunityContainer>
      <p className="title">Communities</p>
      {CommunityList.map((Community) => (
        <CommunityCardItem
          // communityIcon={Community.communityIcon}
          key={Community.communityID}
          communityID={Community.communityID}
          communityDescription={Community.communityDescription}
          membersCount={Community.membersCount}
          communityName={Community.communityName}
        />
      ))}
    </CommunityContainer>
  );
};

export default TopCommunities;
