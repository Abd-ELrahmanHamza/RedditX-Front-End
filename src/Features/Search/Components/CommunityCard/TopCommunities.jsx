import { CommunityContainer } from "./TopCommunities.styled";
// import { useState } from "react";
// import logo from "../../Assets/download.jpg";
import { Link } from "react-router-dom";
import CommunityCardItem from "../CommunityCardItem/CommunityCardItem";
const TopCommunities = () => {
  return (
    <CommunityContainer>
      <p className="title">Communities</p>
      <CommunityCardItem />
      <CommunityCardItem />
      <CommunityCardItem />
      <CommunityCardItem />
      <CommunityCardItem />
      <Link to="/communities">
        <p className="more">See more communities</p>
      </Link>
    </CommunityContainer>
  );
};

export default TopCommunities;
