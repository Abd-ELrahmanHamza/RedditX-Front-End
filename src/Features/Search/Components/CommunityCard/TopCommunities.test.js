import TopCommunities from "./TopCommunities";

import { async } from "@firebase/util";
import { fireEvent, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TestingComponent from "Features/Search/TestingComponent";

const communityList = {
  results: [
    {
      icon: "People_Image.jpg",
      _id: "t5_imagepro",
      description:
        "A subreddit dedicated to German photos and portraits from the period of 1933-1946 (dates are flexible)",
      membersCnt: 10,
    },
  ],
};
const CommunitiesSub2 = [];
describe("TopCommunities component", () => {
  it('renders "Communities" in title', async () => {
    render(
      <TestingComponent>
        <TopCommunities
          CommunityList={communityList}
          CommunitiesSub2={CommunitiesSub2}
        />
      </TestingComponent>
    );
    const outputElement = screen.getByText("Communities");
    expect(outputElement).toBeInTheDocument();
  });
  it('renders "See More Communities" in Card', async () => {
    render(
      <TestingComponent>
        <TopCommunities
          CommunityList={communityList}
          CommunitiesSub2={CommunitiesSub2}
        />
      </TestingComponent>
    );
    const outputElement = screen.getByText("See more communities");
    expect(outputElement).toBeInTheDocument();
  });
  it("check that communities are renders well", async () => {
    render(
      <TestingComponent>
        <TopCommunities
          CommunityList={communityList}
          CommunitiesSub2={CommunitiesSub2}
        />
      </TestingComponent>
    );

    const listItemElements = await screen.findAllByTitle("comm");
    expect(listItemElements).not.toHaveLength(0);
  });
});
// it("this is a test", () => {});
