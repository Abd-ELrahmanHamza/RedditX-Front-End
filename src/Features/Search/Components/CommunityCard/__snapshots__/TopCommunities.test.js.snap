// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Community CardI tem this is a test for Community Card Item 1`] = `ShallowWrapper {}`;

exports[`Top Communities this is a test for Top Communities 1`] = `ShallowWrapper {}`;
